device/mellanox/x86_64-mlnx_lssn2700-r0/plugins/sfputil.py

```diff
@@ -10,6 +10,34 @@
 except ImportError as e:
     raise ImportError("%s - required module not found" % str(e))
 
+def determine_in_docker():
+    with open('/proc/self/cgroup') as f:
+        lines = [line.rstrip('\n') for line in f]
+    for line in lines:
+        words = line.split('/')
+        if len(words) > 1 and words[1] == 'docker':
+            return True
+
+    return False
+
+INSIDE_DOCKER = determine_in_docker()
+
+if INSIDE_DOCKER:
+    from swsscommon import swsscommon
+    REDIS_HOSTNAME = "localhost"
+    REDIS_PORT = 6379
+    REDIS_TIMEOUT_USECS = 0
+
+    state_db = swsscommon.DBConnector(swsscommon.STATE_DB,
+                                     REDIS_HOSTNAME,
+                                     REDIS_PORT,
+                                     REDIS_TIMEOUT_USECS)
+
+    # Subscribe to state table for SFP change notifications
+    sel = swsscommon.Select()
+    sel_tbl = swsscommon.NotificationConsumer(state_db, 'TRANSCEIVER_NOTIFY')
```

Comment:
>TRANSCEIVER_NOTIFY [](start = 57, length = 18)

Just curious which component provide the notification? As [https://github.com/Azure/SONiC/blob/gh-pages/doc/OIDsforSensorandTransciver.MD](https://github.com/Azure/SONiC/blob/gh-pages/doc/OIDsforSensorandTransciver.MD), why let the provide write STATE DB directly?

Reply:
@qiluo-msft on top of the previous comments, on mlnx platform, this notification is exposed from another daemon inside syncd(#https://github.com/Azure/sonic-buildimage/pull/1841),  and that daemon post the notification to the DB. For other vendors, they may have their implementation, for example, get this notification directly from sysfs.